synapse/handlers/sync.py

```diff
@@ -174,37 +194,52 @@ def full_state_sync(self, sync_config, timeline_since_token):
         """
         now_token = yield self.event_sources.get_current_token()
 
-        now_token, ephemeral_by_room = yield self.ephemeral_by_room(
-            sync_config, now_token
-        )
+        if sync_config.is_guest:
+            room_list = [
+                GuestRoom(room_id, Membership.JOIN)
+                for room_id in sync_config.filter.list_rooms()
+            ]
 
-        presence_stream = self.event_sources.sources["presence"]
-        # TODO (mjark): This looks wrong, shouldn't we be getting the presence
-        # UP to the present rather than after the present?
-        pagination_config = PaginationConfig(from_token=now_token)
-        presence, _ = yield presence_stream.get_pagination_rows(
-            user=sync_config.user,
-            pagination_config=pagination_config.get_source_config("presence"),
-            key=None
```

Comment:
I'm not seeing why this code is being removed. It is now:

```python
    presence_stream = self.event_sources.sources["presence"]
    presence, _ = yield presence_stream.get_new_events(
        from_key=0,
        user=sync_config.user,
        room_ids=joined_room_ids,
        is_guest=sync_config.is_guest,
    )
```

which is close but not quite the same. Was this fixing the `TODO`?